src/Compilers/Core/Portable/Operations/IBinaryOperatorExpression.cs

```diff
@@ -32,5 +32,13 @@ public interface IBinaryOperatorExpression : IHasOperatorMethodExpression
         /// value types.
         /// </summary>
         bool IsLifted { get; }
+        /// <summary>
+        /// <code>true</code> if this is a 'checked' binary operator.
+        /// </summary>
+        bool IsChecked { get; }
```

Comment:
What about things where IsChecked doesn't make sense, like all statements? The Type property at least has a way to show that Type isn't relevant for the operation, by returning null. Seems like we wouldn't be able to distinguish between "This isn't a checked expression" and "IsChecked has no meaning here" easily.